### PR TITLE
**6X Backport** CI: unvendor behave and install patchelf

### PR DESCRIPTION
**Pull request by kalensk**
This is a more robust solution to https://github.com/greenplum-db/gpdb/pull/8138 which addressed an additional Behave1.2.6 transitive dependency. This PR unvendors behave, and introduces pip to better enable python dependency management. It is a manual cherry-pick of PR https://github.com/greenplum-db/gpdb/pull/7503 which outlines in detailed everything that is needed.

### CODE REVIEW

File: concourse/scripts/common.bash [hunk 1]
```diff
@@ -60,3 +60,89 @@ function run_test() {
...
+
+    # Install requirements into the vendored Python stack on all hosts.
+    mkdir -p /tmp/py-requirements
+    source /tmp/venv/bin/activate
+        pip install --prefix /tmp/py-requirements -r ${requirements_txt}
+        while read -r host; do
+            rsync -rz /tmp/py-requirements/ "$host":/usr/local/greenplum-db-devel/ext/python/
+        done < /tmp/hostfile_all
+    deactivate
+}
```
Comment (dkrieger-pvtl) [inline concourse/scripts/common.bash]: supernit: add a new-line here

**Comment by dkrieger-pvtl** [comment]
It's not part of this PR, but for our future selves, I wonder how many of the other vendored libraries(PARSE, etc) we can nix since we are unvendoring behave.

**Comment by dkrieger-pvtl** [comment]
After this PR is merged and considered successful, we should revert the previous, expedient fix to this issue: https://github.com/greenplum-db/gpdb/pull/7503.

File: gpMgmt/bin/Makefile [hunk 2]
```diff
@@ -144,19 +144,16 @@ LOGILAB_COMMON_DIR=logilab-common-$(LOGILAB_COMMON_VERSION)
 PYLINT_PYTHONPATH=$(PYLIB_DIR):$(PYLIB_SRC_EXT)/$(PYLINT_DIR)/build/lib/
 MOCK_VERSION=1.0.1
 MOCK_DIR=mock-$(MOCK_VERSION)
-BEHAVE_VERSION=1.2.6
 SETUP_TOOLS_VERSION=36.6.0
 PARSE_VERSION=1.8.4
```
Comment (dkrieger-pvtl) [inline gpMgmt/bin/Makefile]: set `PARSE_VERSION=1.8.2` to "revert" 3150dff54d and prove that we are, in fact, resolving the issue.

**Comment by jmcatamney** [comment]
> One question was why master has(from `concourse/scripts/run_behave_test.sh`) this snippet whereas 6X_STABLE(and hence this PR) doesn't. Presumably it is for code coverage but I'm not sure.
> 
> ```
> CWDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
> source "${CWDIR}/common.bash"
> ```

Yep, that's for code coverage, see commit 97bcd34591d3ed2eb07e685880199b093913af5e.

**Comment by jmcatamney** [comment]
> One question was why master has(from `concourse/scripts/run_behave_test.sh`) this snippet whereas 6X_STABLE(and hence this PR) doesn't. Presumably it is for code coverage but I'm not sure.
> 
> ```
> CWDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
> source "${CWDIR}/common.bash"
> ```

Yep, that's for code coverage, see commit 97bcd34591d3ed2eb07e685880199b093913af5e.

**Pull request merged**